pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -505,15 +505,15 @@ func TestPreFilterState(t *testing.T) {
 					{
 						MaxSkew:            3,
 						TopologyKey:        "node",
-						Selector:           mustConvertLabelSelectorAsSelector(t, st.MakeLabelSelector().Label("foo", "bar").Obj()),
+						Selector:           labels.SelectorFromValidatedSet(labels.Set{"foo": "bar"}),
 						MinDomains:         1,
 						NodeAffinityPolicy: v1.NodeInclusionPolicyHonor,
 						NodeTaintsPolicy:   v1.NodeInclusionPolicyIgnore,
 					},
 					{
 						MaxSkew:            5,
 						TopologyKey:        "rack",
-						Selector:           mustConvertLabelSelectorAsSelector(t, st.MakeLabelSelector().Label("foo", "bar").Obj()),
+						Selector:           labels.SelectorFromValidatedSet(labels.Set{"foo": "bar"}),
```

Comment:
These tests depended on internal impl details by using reflection to compare unexported fields, so needed to be updated

Reply:
rather than switching the construction, shouldn't we switch the comparison to compare selectors more accurately (e.g. compare `a.String() == b.String()`)?